32_awk_lab/awklab.txt

```diff
@@ -20,6 +21,7 @@ p2p0: flags=8843 mtu 2304
     status: inactive
 
 Write an awk one liner that splits this file into multiple file, one for each block, so after running the script we'll have a file named "en3" with the lines of the first block, and another file named "p2p0" with the lines of the second block.
+Answer: /sbin/ifconfig | awk ' BEGIN { block=1 } { if(length($0) > 0) print $0 > block } { if(length($0) == 0) block++ } '
```

Comment:
hmm.. almost! you need to make sure the files are named after the the interface, that's actually alot easier than it looks:
```awk '/^[a-z]/ {file_name=$1 } {print $0 >> file_name}'```


```/^[a-z]/ {file_name=$1 }``` - match any line that starts with lower case letter and set ```file_name``` to the first word for that line

```{print $0 >> file_name}``` - match **everything**, and print it to a file called ```file_name```

if it's not clear, try this:
```ifconfig | awk '/^[a-z]/ {file_name=$1 } {print file_name ":" $0}'```